src/main/groovy/com/contrastsecurity/InstallContrastAgent.groovy

```diff
@@ -50,6 +68,27 @@ class InstallContrastAgent extends DefaultTask {
             if (!argLineContainsContrastArgLine(property)) {
                 config.setProperty(jvmArgs, property + " " + ContrastGradlePlugin.buildArgLine(project))
                 layout.save(new FileWriter(gradleProperties, false))
+            } else {
```

Comment:
I admire what you're trying to do here but it's getting a little hard to follow. For the sake of saving us from all this complexity, let's just do this:

if `argLineContainsContrastArgLine` is true, then just do nothing. This means the user specified their own contrast argLine. We're not going to mess with it.

Reply:
@gmoore Currently the appVersionQualifier is generated each time the install task is run, so it will be different from the one in argline if we don't update it each time. Should we take it from the argline instead if its present?

Reply:
I don't understand that the argLine persists between runs but the appVersionQualifier does not.

Reply:
@gmoore Now, appVersionQualifier is not changed if it is already set.